### ISSUE TITLE
Issues about multi-shell

### ISSUE THREAD

**Issue by XiZhu-UE** [m-lyon/dMRI-RCNN#8]
Dear  Matthew Lyon  ,

I am currently working on super resolution for dMRI, and I came across some issues while using/understanding your code named 3D RCNN.

Specifically, I am having difficulty with running inference code.  When I read your NIPS paper, I noticed that you compared RCNN in a multi-shell scheme (b1000 as input, b1000,2000,3000 as output). But there are no public weights for this and when I try to accomplish this I do not know how to process target bvecs. So, whether you can give me weights and code about this or you can tell me how to process target bvecs.  I would greatly appreciate it if you could provide some guidance or clarification on this matter.

Thank you very much for your time and assistance. I look forward to your reply.

Best regards, Your research inspired me so much!

**Comment by m-lyon** [comment]
Hey there,

Thank you for your kind words. There are public weights for the b1000 as input are available, depending on the `q_in` size ($q_{\mathrm{in}} \in {6, 10, 30}$ are available). The code within [weights.py](https://github.com/m-lyon/dMRI-RCNN/blob/main/dmri_rcnn/core/weights.py) point to the model weights. All of the models were trained in a multi-shell regime so can predict any of the three shells (b = 1000, 2000, 3000). Hope this helps, and if you have any other questions please feel free to ask.

**Comment by XiZhu-UE** [comment]
Thanks for your reply! You mean that when I use the model "q-in is 6-dir, dim is 3D, shell is b1000", this model can predict b2000 and b3000 shells' data?

**Comment by XiZhu-UE** [comment]
Perhaps I haven't expressed myself clearly. The model I want to use takes b1000 data in six directions as input and any direction of b1000/b2000/b3000 as output (the model can predict b2000 and b3000 data from the b1000 data). Your code seems to require that the input and output be the same shell.

**Comment by m-lyon** [comment]
Ah, my mistake. I've worked on this problem in a couple different codebases and I didn't realise this issue was raised in the earlier work. This project, dMRI-RCNN, indeed was not trained in a multi-shell setting. A later project, [dmri-pcconv](https://github.com/m-lyon/dmri-pcconv) does train a model in a multi-shell setting. And in fact I reimplemented the dMRI-RCNN model in PyTorch in that project (I did train the dMRI-RCNN in a multishell setting there but didn't upload the weights -  let me know if you would like them and I can set aside some time to upload them for the PyTorch implementation).

**Comment by XiZhu-UE** [comment]
I would be extremely grateful if you could upload the weights and code for that version! That would be a great help to me!

**Comment by m-lyon** [comment]
I've found the weights, however the results from using the same preprocessing code as with the main model in that project (the PCCNN), leads to incorrect output. I'm not really sure where the issue is unfortunately.

Nonetheless, I've uploaded the weights for the RCNN [here](https://zenodo.org/records/17506562) and [here](https://zenodo.org/records/17506504) that should be used with the [RCNN torch module](https://github.com/m-lyon/dmri-pcconv/blob/99416289d14b2c9e54bc4220d40fbaa6784a42c3/dmri_pcconv/core/model/rcnn.py#L142). Feel free to try and get it to work, but unfortunately the best I can offer you at this time for working multi-shell prediction is the PCCNN models available in that project.
